lib/bridge/matrix-handler.js

```diff
@@ -568,25 +584,78 @@ let _onAliasQuery = Promise.coroutine(function*(req, roomAlias) {
     );
 
     if (matrixRooms.length === 0) {
+        // ====== Track the IRC channel
         // lower case the name to join (there's a bug in the IRC lib
         // where the join callback never fires if you try to join
         // #WithCaps in channels :/)
         channelInfo.channel = toIrcLowerCase(channelInfo.channel);
-        req.log.info("Going to track channel %s", channelInfo.channel);
+        req.log.info("Going to track IRC channel %s", channelInfo.channel);
         // join the irc server + channel
         yield ircLib.trackChannel(channelInfo.server, channelInfo.channel);
-        req.log.info("Bot is now tracking channel.");
-        let matrixRoom = yield req.mxLib.createRoomWithAlias(
-            roomAlias, channelInfo.channel, undefined,
-            channelInfo.server.getJoinRule(),
-            channelInfo.server.shouldFederate(),
-            channelInfo.server.shouldPublishRooms()
-        );
+        req.log.info("Bot is now tracking IRC channel.");
+
+        // ======== Create the Matrix room
+        let newRoomId = null;
+        let botIntent = this.ircBridge.bridge.getIntent();
+        try { // make the matrix room
```

Comment:
Drop the scope of your try to the things which could actually fail (i.e. the createRoom call, and if they remain, the sendStateEvent calls)